codelab-master/apps/playground/src/app/quiz/containers/scoreboard/time/time.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { concat, Observable, timer } from 'rxjs';
+import { first, repeatWhen, scan, shareReplay, skip, switchMapTo, takeUntil, tap } from 'rxjs/operators';
+
+import { QuizService, TimerService } from '@codelab-quiz/shared/services/*';
+
+@Component({
+  selector: 'codelab-scoreboard-time',
+  templateUrl: './time.component.html',
+  styleUrls: ['./time.component.scss']
+})
+export class TimeComponent implements OnChanges {
+  @Input() set selectedAnswer(value) { this.answer = value; }
+  answer;
+  timeLeft$: Observable<number>;
+  timePerQuestion = 20;
+
+  constructor(
+    private quizService: QuizService,
+    private timerService: TimerService
+  ) {
+    this.countdownClock();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (
+      changes.selectedAnswer &&
+      changes.selectedAnswer.currentValue !== changes.selectedAnswer.firstChange
+    ) {
+      this.answer = changes.selectedAnswer.currentValue;
+    }
+  }
+
+  countdownClock(): void {
+    const start$ = this.timerService.isStart.asObservable().pipe(shareReplay(1));
```

Comment:
You shouldn't be exposing subjects from the service, do asObservable there, and expose that instead

Reply:
ok will do :)

Reply:
OK done!